src/saveToCase.js

```diff
@@ -0,0 +1,475 @@
+                }
+                return ret;
+            },
+            parseDataNode: function (mug, $node) {
+                var case_ = $node.children(),
+                    create = case_.find('create'),
+                    close = case_.find('close'),
+                    update = case_.find('update');
+                if (create && create.length !== 0) {
+                    mug.p.use_create = true;
+                }
+                if (update && update.length !== 0) {
+                    mug.p.use_update = true;
+                }
+                if (close && close.length !== 0) {
+                    mug.p.use_close = true;
+                }
+                return $([]);
+            }
+        },
+        sectionData = {
+            SaveToCase: [
+                {
+                    slug: "main",
+                    displayName: "Basic",
+                    properties: [
+                        "nodeID",
+                        "date_modified",
+                        "user_id",
+                        "case_id",
+                    ],
+                },
+                {
+                    slug: "create",
+                    displayName: "Create",
+                    properties: [
+                        "use_create",
+                        "create_property",
+                    ],
+                    isCollapsed: function (mug) {
+                        return !createsCase(mug);
+                    },
+                },
+                {
+                    slug: "update",
+                    displayName: "Update",
+                    properties: [
+                        "use_update",
+                        "update_property",
+                    ],
+                    isCollapsed: function (mug) {
+                        return !updatesCase(mug);
+                    },
+                },
+                {
+                    slug: "close",
+                    displayName: "Close",
+                    properties: [
+                        "use_close",
+                        "close_condition",
+                    ],
+                    isCollapsed: function (mug) {
+                        return !closesCase(mug);
+                    },
+                }
+            ]
+        };
+
+    $.vellum.plugin("saveToCase", {}, {
+        getAdvancedQuestions: function () {
+            return this.__callOld().concat(["SaveToCase"]);
+        },
+        handleMugParseFinish: function (mug) {
+            this.__callOld();
+            // cases that are created use a setvalue for case_id
+            if (!createsCase(mug)) {
+                return;
+            }
+            var values = _.object(_.map(mug.form.getSetValues(), function (value) {
+                    return [value.ref, value];
+                }));
+
+            _.each(values, function(value) {
+                if (/case\/@case_id$/.test(value.ref)) {
+                    mug.p.case_id = value.value;
```

Comment:
What if there is more than one SaveToCase property in the form? Will this end up setting the case_id of this mug to the last setvalue found with a path like `...case/@case_id`?